tests/unit/test_reconcile.py

```diff
@@ -0,0 +1,709 @@
+    navlink = types_.Navlink(title=navlink_title, link=(navlink_link := "link 1"))
+    table_row = types_.TableRow(level=level, path=table_path, navlink=navlink)
+
+    returned_action = reconcile._local_and_server(
+        path_info=path_info, table_row=table_row, discourse=mock_discourse
+    )
+
+    assert returned_action.action == types_.Action.UPDATE
+    assert returned_action.level == level
+    assert returned_action.path == table_path
+    # mypy has difficulty with determining which action is returned
+    assert returned_action.navlink_change.old == navlink  # type: ignore
+    assert returned_action.navlink_change.new == navlink  # type: ignore
+    assert returned_action.content_change.old == server_content  # type: ignore
+    assert returned_action.content_change.new == local_content  # type: ignore
+    mock_discourse.retrieve_topic.assert_called_once_with(url=navlink_link)
+
+
+def test__local_and_server_file_navlink_title_change(tmp_path: Path):
+    """
+    arrange: given path info with a file and table row with different navlink title and discourse
+        client that returns the same content as in the file
+    act: when _local_and_server is called with the path info and table row
+    assert: then an update action is returned.
+    """
+    (path := tmp_path / "file1.md").touch()
+    path.write_text(content := "content 1", encoding="utf-8")
+    path_info = types_.PathInfo(
+        local_path=path,
+        level=(level := 1),
+        table_path=(table_path := "table path 1"),
+        navlink_title=(local_navlink_title := "title 1"),
+    )
+    mock_discourse = mock.MagicMock(spec=discourse.Discourse)
+    mock_discourse.retrieve_topic.return_value = content
+    navlink = types_.Navlink(title="title 2", link=(navlink_link := "link 1"))
+    table_row = types_.TableRow(level=level, path=table_path, navlink=navlink)
+
+    returned_action = reconcile._local_and_server(
+        path_info=path_info, table_row=table_row, discourse=mock_discourse
+    )
+
+    assert returned_action.action == types_.Action.UPDATE
+    assert returned_action.level == level
+    assert returned_action.path == table_path
+    # mypy has difficulty with determining which action is returned
+    assert returned_action.navlink_change.old == navlink  # type: ignore
+    assert returned_action.navlink_change.new == types_.Navlink(  # type: ignore
+        title=local_navlink_title, link=navlink_link
+    )
+    assert returned_action.content_change.old == content  # type: ignore
+    assert returned_action.content_change.new == content  # type: ignore
+    mock_discourse.retrieve_topic.assert_called_once_with(url=navlink_link)
+
+
+def test__local_and_server_directory_same(tmp_path: Path):
+    """
+    arrange: given path info with a directory and table row with no changes
+    act: when _local_and_server is called with the path info and table row
+    assert: then a noop action is returned.
+    """
+    (path := tmp_path / "dir1").mkdir()
```

Comment:
Are the temp files and directories cleaned up and do we want to clean them up? Could something like `tempfile` be used here?

Reply:
`temp_file` is a fixture provided and managed by `pytest`.

Reply:
`tmp_path` is a fixture provided by `pytest` which takes care of creation and cleanup: https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

Reply:
Just closing this since I believe I addressed it, please let me know if you have further feedback